concordium-sdk/src/main/java/com/concordium/sdk/cis2/events/MintEvent.java

```diff
@@ -0,0 +1,43 @@
+package com.concordium.sdk.cis2.events;
+
+import com.concordium.sdk.cis2.TokenId;
+import com.concordium.sdk.types.AbstractAddress;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+/**
+ * An event emitted by a CIS2 compliant smart contract when one or more tokens have been minted.
+ * https://proposals.concordium.software/CIS/cis-2.html#mintevent
+ */
+@Getter
+@ToString
+@EqualsAndHashCode
+public class MintEvent implements Cis2Event {
+
+    /**
+     * ID of the token that was minted.
+     */
+    private final TokenId tokenId;
+
+    /**
+     * Amount of tokens that was minted.
+     */
+    private final long tokenAmount;
```

Comment:
Amounts are not `long`. They can be up to 256 bits.

Reply:
Same in other places.

Reply:
Introduced a `TokenAmount` class which aims to fix the comments you've had.